crates/forge_services/src/tool_services/shell.rs

```diff
@@ -47,10 +47,14 @@ impl<I: CommandInfra + EnvironmentInfra> ShellService for ForgeShell<I> {
         command: String,
         cwd: PathBuf,
         keep_ansi: bool,
+        channel: &mut (impl WriteChannel + Send + Sync),
     ) -> anyhow::Result<ShellOutput> {
         Self::validate_command(&command)?;
 
-        let mut output = self.infra.execute_command(command, cwd).await?;
+        let mut output = self
+            .infra
+            .execute_command_streaming(command, cwd, channel)
+            .await?;
```

Comment:
We should use a feature flag to switch between the two implementations `execute_command` and `execute_command_streaming`. This should be a feature flag.